src/Library/demos/File Monitor/main.js

```diff
@@ -0,0 +1,75 @@
+import Gtk from "gi://Gtk";
+import Gio from "gi://Gio";
+import Adw from "gi://Adw";
+import GLib from "gi://GLib";
+
+Gio._promisify(Gtk.FileLauncher.prototype, "launch", "launch_finish");
+
+const edit_entry = workbench.builder.get_object("edit_entry");
+const view_file = workbench.builder.get_object("view_file");
+const create_file = workbench.builder.get_object("create_file");
+const delete_file = workbench.builder.get_object("delete_file");
+const edit_file = workbench.builder.get_object("edit_file");
+const file_name = workbench.builder.get_object("file_name");
+const { buffer } = edit_entry;
+const file = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
+  "Library/demos/File Monitor/workbench.txt",
+);
+const file_dir = file.get_parent();
+const file_launcher = new Gtk.FileLauncher({
+  always_ask: true,
+  file,
+});
+const monitor_for_dir = file.monitor(Gio.FileMonitorFlags.NONE, null);
```

Comment:
```suggestion
const monitor_for_dir = file_dir.monitor(Gio.FileMonitorFlags.WATCH_MOVES, null);
```

Don't forget to set the correct flags for what you want to monitor.